src/libstd/macros.rs

```diff
@@ -155,8 +155,14 @@ macro_rules! print {
 #[stable(feature = "rust1", since = "1.0.0")]
 macro_rules! println {
     () => (print!("\n"));
-    ($fmt:expr) => (print!(concat!($fmt, "\n")));
-    ($fmt:expr, $($arg:tt)*) => (print!(concat!($fmt, "\n"), $($arg)*));
+    ($fmt:expr) => ({
+        print!($fmt);
+        print!("\n");
```

Comment:
Using two `print!`s will mean stdout will be unlocked before printing the `\n`. Replacing these two rules with

```rust
($($arg:tt)*) => (print!("{}\n", format_args!($($arg)*)));
```

will avoid that issue.